src/read.h

```diff
@@ -27,18 +27,30 @@
 
 #ifndef DEBUG_READ_ERROR
 
-#define READ_ERROR()    (TLS(NrError) || (TLS(NrError)+=sySetjmp(TLS(ReadJmpError))))
+#define TRY_READ \
+    if (!TLS(NrError)) { \
+        if (sySetjmp(TLS(ReadJmpError))) { \
+            TLS(NrError)++; \
+        }\
+    }\
+    if (!TLS(NrError))
 
 #else
 
-#define READ_ERROR()                                                     \
-    ( TLS(NrError) ||                                                         \
-      ( ( TLS(NrError) += setjmp(TLS(ReadJmpError)) ) ?                            \
-        Pr( "READ_ERROR( %s, %d )\n", (Int)__FILE__, __LINE__ ),0 : 0 ), \
-      TLS(NrError) )
+#define TRY_READ \
+    if (!TLS(NrError)) { \
+        if (sySetjmp(TLS(ReadJmpError))) { \
+            Pr( "READ_ERROR( %s, %d )\n", (Int)__FILE__, __LINE__ ); \
+            TLS(NrError)++; \
+        }\
+    }\
+    if (!TLS(NrError)) {
 
 #endif
 
+#define CATCH_READ_ERROR \
+    else
+
```

Comment:
Yes, there is a reason: It hides away the details of the implementation of `TRY_READ`. E.g. it allows us to insert a `memcpy(old_jumpbuf, current_jumpbuf, ...)` to support chained TRY.
Also, somebody not familiar with this might be confused by the `else` -- such as syntax highlighters ;-).

(Indeed, one major drawback of this preprocessor trick is that code formatters won't know how to deal with it suitably. But I can live with that).